main.tf

```diff
@@ -1,16 +1,151 @@
+data "aws_caller_identity" "current" {
+}
 
-module "tags" {
-  source  = "rhythmictech/tags/terraform"
-  version = "1.0.0"
-
-  enforce_case = "UPPER"
-  names        = [var.name]
-  tags         = var.tags
+data "aws_region" "current" {
 }
 
 locals {
-  # tflint-ignore: terraform_unused_declarations
-  name = module.tags.name
-  # tflint-ignore: terraform_unused_declarations
-  tags = module.tags.tags_no_name
+  account_id   = data.aws_caller_identity.current.account_id
+  bucket_name  = var.bucket_name == null ? "${local.account_id}-${local.region}-cloudberry-${var.bucket_suffix}" : var.bucket_name
+  iam_role_arn = var.iam_role_arn == null ? aws_iam_role.this[0].id : var.iam_role_arn
+  kms_key_id   = var.kms_key_id == null ? "aws/s3" : var.kms_key_id
+  region       = data.aws_region.current.name
+
+  logging = var.s3_access_logging_bucket == null ? [] : [{
+    bucket = var.s3_access_logging_bucket
+    prefix = var.s3_access_logging_prefix
+  }]
+}
+
+#tfsec:ignore:AWS002
+resource "aws_s3_bucket" "this" {
+  bucket = local.bucket_name
+  acl    = "private"
+  tags   = var.tags
+
+  dynamic "lifecycle_rule" {
+    iterator = rule
+    for_each = var.lifecycle_rules
+
+    content {
+      id      = rule.value.id
+      enabled = rule.value.enabled
+      prefix  = lookup(rule.value, "prefix", null)
+
+      expiration {
+        days = rule.value.expiration
+      }
+
+      noncurrent_version_expiration {
+        days = rule.value.noncurrent_version_expiration
+      }
+    }
+  }
+
+  dynamic "logging" {
+    iterator = log
+    for_each = local.logging
+
+    content {
+      target_bucket = log.value.bucket
+      target_prefix = lookup(log.value, "prefix", null)
+    }
+  }
+
+  server_side_encryption_configuration {
+    rule {
+      apply_server_side_encryption_by_default {
+        sse_algorithm     = "aws:kms"
+        kms_master_key_id = local.kms_key_id
+      }
+    }
+  }
+
+  versioning {
+    enabled = var.versioning_enabled
+  }
+
+  # this cannot be configured programatically via TF, so just ignore it if someone
+  # turned it on administratively.
```

Comment:
Does this sentence need a period? When I write comments I always second guess this. IDK if there's a good argument for either side